components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1262,22 +1298,18 @@ void KeyringService::AddHardwareAccounts(
           kAccountMetas, base::Value(base::Value::Type::DICTIONARY));
     }
 
-    DCHECK_EQ(hardware_vendor, info->hardware_vendor);
-    if (hardware_vendor != info->hardware_vendor)
-      continue;
-    base::Value hw_account(base::Value::Type::DICTIONARY);
-    hw_account.SetStringKey(kAccountName, info->name);
-    hw_account.SetStringKey(kHardwareVendor, info->hardware_vendor);
-    hw_account.SetStringKey(kHardwareDerivationPath, info->derivation_path);
-
     meta_value->SetKey(info->address, std::move(hw_account));
   }
 
   NotifyAccountsChanged();
 }
 
 void KeyringService::RemoveHardwareAccount(const std::string& address) {
-  base::Value* hardware_keyrings = GetPrefForHardwareKeyringUpdate(prefs_);
+  auto keyring_id = GetKeyringIdForHardwareAccount(address);
```

Comment:
Do not iterate all keyrings for this info, caller already has the information of which CoinType this account belongs.
see https://github.com/brave/brave-core/pull/12282/files#diff-af0cc8816d323a308707bc347f56b9368f8b5e05152174c35b07068bb88908c9R433-R438

Reply:
great, will remove this after rebase, thanks

Reply:
done